Validation/HGCalValidation/plugins/HGCGeometryValidation.cc

```diff
@@ -201,87 +198,84 @@ void HGCGeometryValidation::analyze(const edm::Event &iEvent, const edm::EventSe
-                   (hitDet.at(i) == (unsigned int)(DetId::HGCalHSi))) {
-          HGCSiliconDetId id(hitIdx.at(i));
-          dtype = (id.det() == DetId::HGCalEE) ? 0 : 1;
-          layer = id.layer();
-          zside = id.zside();
-          xy = hgcGeometry_[dtype]->locateCell(layer, id.waferU(), id.waferV(), id.cellU(), id.cellV(), true, true);
-        } else {
-          HGCScintillatorDetId id(hitIdx.at(i));
-          dtype = 2;
-          layer = id.layer();
-          zside = id.zside();
-          xy = hgcGeometry_[dtype]->locateCellTrap(layer, id.ietaAbs(), id.iphi(), true);
-        }
-        double zz = hgcGeometry_[dtype]->waferZ(layer, true);  //cm
-        if (zside < 0)
-          zz = -zz;
-        xx = (zside < 0) ? -xy.first : xy.first;
-        yy = xy.second;
-
-        if (dtype == 0) {
-          heedzVsZ->Fill(zz, (hitVtxZ.at(i) - zz));
-          heedyVsY->Fill(yy, (hitVtxY.at(i) - yy));
-          heedxVsX->Fill(xx, (hitVtxX.at(i) - xx));
-
-          heeXG4VsId->Fill(hitVtxX.at(i), xx);
-          heeYG4VsId->Fill(hitVtxY.at(i), yy);
-          heeZG4VsId->Fill(hitVtxZ.at(i), zz);
-
-          heedzVsLayer->Fill(layer, (hitVtxZ.at(i) - zz));
-          heedyVsLayer->Fill(layer, (hitVtxY.at(i) - yy));
-          heedxVsLayer->Fill(layer, (hitVtxX.at(i) - xx));
-
-          heedX->Fill((hitVtxX.at(i) - xx));
-          heedZ->Fill((hitVtxZ.at(i) - zz));
-          heedY->Fill((hitVtxY.at(i) - yy));
-
-        } else if (dtype == 1) {
-          hefdzVsZ->Fill(zz, (hitVtxZ.at(i) - zz));
-          hefdyVsY->Fill(yy, (hitVtxY.at(i) - yy));
-          hefdxVsX->Fill(xx, (hitVtxX.at(i) - xx));
-
-          hefXG4VsId->Fill(hitVtxX.at(i), xx);
-          hefYG4VsId->Fill(hitVtxY.at(i), yy);
-          hefZG4VsId->Fill(hitVtxZ.at(i), zz);
-
-          hefdzVsLayer->Fill(layer, (hitVtxZ.at(i) - zz));
-          hefdyVsLayer->Fill(layer, (hitVtxY.at(i) - yy));
-          hefdxVsLayer->Fill(layer, (hitVtxX.at(i) - xx));
-
-          hefdX->Fill((hitVtxX.at(i) - xx));
-          hefdZ->Fill((hitVtxZ.at(i) - zz));
-          hefdY->Fill((hitVtxY.at(i) - yy));
-
-        } else {
-          hebdzVsZ->Fill(zz, (hitVtxZ.at(i) - zz));
-          hebdyVsY->Fill(yy, (hitVtxY.at(i) - yy));
-          hebdxVsX->Fill(xx, (hitVtxX.at(i) - xx));
-
-          hebXG4VsId->Fill(hitVtxX.at(i), xx);
-          hebYG4VsId->Fill(hitVtxY.at(i), yy);
-          hebZG4VsId->Fill(hitVtxZ.at(i), zz);
-
-          hebdzVsLayer->Fill(layer, (hitVtxZ.at(i) - zz));
-          hebdyVsLayer->Fill(layer, (hitVtxY.at(i) - yy));
-          hebdxVsLayer->Fill(layer, (hitVtxX.at(i) - xx));
-
-          hebdX->Fill((hitVtxX.at(i) - xx));
-          hebdZ->Fill((hitVtxZ.at(i) - zz));
-          hebdY->Fill((hitVtxY.at(i) - yy));
-        }
+      double xx, yy;
+      int dtype(0), layer(0), zside(1);
+      std::pair<float, float> xy;
+      if (hitDet.at(i) == static_cast<unsigned int>(DetId::Forward)) {
```

Comment:
* Using `[]` instead of `at()` is maybe faster since it does not have the internal boundary check.
* Also a few lines above [166-175] (not part of this PR's changes): are we sure we want to copy all vectors? Is it maybe possible to have a reference to the original vectors?
* IIRC, the `DetId::Forward` is no longer used in any recent geometry (or at least in the ones we maintain as active in the release). Is that correct? If so, we could remove this section.